### DISCUSSION TITLE
Can we get the raw string in the MQTT message as well?

### DISCUSSION THREAD

**Discussion by zelbanna** [UtilitechAS/amsreader-firmware#887] [Ideas]
Hi

I am feeding reader data to Home Assistant - I would however like to also feed the same data (HAN port so multiple lists/payloads I guess) to Tibber so that they can feed loadshare information to Easee charing box. I know I could monitor the reader feed and republish the information but I haven't found any guide on how to encode DLMS :-)

Would that be possible (either as double publish or as a raw-data attribute)?

**Comment by ArnieO** [comment]
You want the DLMS raw data string in MQTT so that you can decode it yourself?
Are you sure you want to go there? 😉 
This seems like a very "narrow" request, I'm not sure if we want to do that - for several reasons. 

Why not split the HAN port signal, and feed it to Tibber Pulse and Amsreader devices in parallell? If you do that, you will have to power both devices from USB - but appears like an easier solution.

**Reply by zelbanna** [reply]
Almost right :-) - actually I want to make a simple MQTT subscriber that essentially bridge the raw string to Tibber (so pull the raw strings and add some extra messages to mimic the pulse's behavior). I don't have any power outlet close by the meter location (yes for real)  so external feed is no go and Pulse is so unreliable wrt to wifi (I have to store it in the cabinet to avoid rain etc) which is the reason going with the reliable Pow-K+ in the first place.

The other option is obviously to subscribe AMSreader message and craft the DLMS packet myself, and third would be just pull data from AMS reader in raw format and go from there for HA...

If I could get the string in addition to the decode then this would be a simple copy paste operation for the subscriber :-D

**Reply by zelbanna** [reply]
But I hear you - pretty narrow (but I was thinking that if you anyway decoded the string a simple copy of the original into an attribute wouldn't be that hard)

Have a nice evening!

**Reply by ArnieO** [reply]
OK - I see your problem. But again: Seems to be a too "narrow" request. 
Thank you for your understanding!
